client/src/components/page/Sidebar/index.tsx

```diff
@@ -71,25 +74,33 @@ export const SidebarComponent: React.FC<SidebarProps> = (props: SidebarProps) =>
           <img src={Logo} className="img-fluid" alt="" />
         </Link>
       </div>
-      <Menu theme="dark" defaultSelectedKeys={['1']} mode="inline" className="menu-dark">
+
+      <Menu
+        theme="dark"
+        selectedKeys={[loc[loc.length - 1]]}
```

Comment:
@haseemVD would using the full `location.pathname` work here? i.e. changing the keys below to also be same as the pathnames.

I feel like the current approach is coupled to the pathnames in a way that dictates/places constraints on how a route pathname should look like, which I do not think is something we necessarily want.

Reply:
We have updated our urls in such a way that they dictates the hierarchy of menu items,
And `selectedKeys` requires and array of strings from upper level to level.

The implementation for routes was discussed with @KipSigei ,
If you have any suggestions about how we can make this dynamic, then that would be good.

Reply:
if you have 2 routes like:
- `plans/drafts`
- `products/drafts`
would there be collision in here?

Reply:
In case of `plans/drafts`. Plans tab will be expanded first then draft. if exists.
same for `products/drafts`.

Shortly, URL pattern and menu items should match, menu items depends on it.

Reply:
In case of `plans/drafts`. Plans tab will be expanded first then draft. if exists.
same for `products/drafts`.

Shortly, URL pattern and menu items should match, menu items depends on it.

Reply:
ok, got it.

Reply:
one final concern, given the above routes, it seems that we could end up with the a menu Item sharing keys, if this is correct how do we prevent that

Reply:
If we have 2 identical keys in different menus, selection will case an issue.
e.g if we have two menu `plans/drafts` and `products/drafts`
User selects `plans/drafts` then `products/drafts` will also be highlighted.

Collision is not dependent on this.

Reply:
Created issue #304  for suggestions.